app/upgrades/v7/upgrades.go

```diff
@@ -0,0 +1,199 @@
+		CurrentEpoch:            0,
+		CurrentEpochStartHeight: 0,
+		CurrentEpochStartTime:   time.Time{},
+		EpochCountingStarted:    false,
+	}
+
+	k.SetEpochInfo(ctx, hourEpoch)
+}
+
+// Increase stride inflation from 285 STRD to 1078 STRD
+func IncreaseStrideInflation(ctx sdk.Context, k mintkeeper.Keeper) {
+	ctx.Logger().Info("Increasing STRD inflation")
+
+	epochProvisions := sdk.NewDec(1_078_767_123)
+	minter := minttypes.NewMinter(epochProvisions)
+	k.SetMinter(ctx, minter)
+}
+
+// Add stride messages (LiquidStake, RedeemStake, Claim) to the ICAHost allow messages
+// to allow other protocols to liquid stake via ICA
+func AddICAHostAllowMessages(ctx sdk.Context, k icahostkeeper.Keeper) {
+	ctx.Logger().Info("Adding ICA Host allow messages")
+
+	params := icahosttypes.Params{
+		HostEnabled: true,
+		AllowMessages: []string{
+			sdk.MsgTypeURL(&banktypes.MsgSend{}),
+			sdk.MsgTypeURL(&banktypes.MsgMultiSend{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgDelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgUndelegate{}),
+			sdk.MsgTypeURL(&stakingtypes.MsgBeginRedelegate{}),
+			sdk.MsgTypeURL(&distrtypes.MsgWithdrawDelegatorReward{}),
+			sdk.MsgTypeURL(&distrtypes.MsgSetWithdrawAddress{}),
+			sdk.MsgTypeURL(&transfertypes.MsgTransfer{}),
+			sdk.MsgTypeURL(&govtypes.MsgVote{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgLiquidStake{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgRedeemStake{}),
+			sdk.MsgTypeURL(&stakeibctypes.MsgClaimUndelegatedTokens{}),
+		},
+	}
+	k.SetParams(ctx, params)
+}
+
+// Add the min/max redemption rate to each host zone as safety bounds, using the default for each
+// Also set the Halted boolean to false
+func AddRedemptionRateSafetyChecks(ctx sdk.Context, k stakeibckeeper.Keeper) {
+	ctx.Logger().Info("Setting min/max redemption rate safety bounds on each host zone")
+
+	// Get default min/max redemption rate
+	defaultMinRedemptionRate := sdk.NewDecWithPrec(int64(k.GetParam(ctx, types.KeyDefaultMinRedemptionRateThreshold)), 2)
+	defaultMaxRedemptionRate := sdk.NewDecWithPrec(int64(k.GetParam(ctx, types.KeyDefaultMaxRedemptionRateThreshold)), 2)
+
+	for _, hostZone := range k.GetAllHostZone(ctx) {
+
+		hostZone.MinRedemptionRate = defaultMinRedemptionRate
+		hostZone.MaxRedemptionRate = defaultMaxRedemptionRate
+		hostZone.Halted = false
+
+		k.SetHostZone(ctx, hostZone)
+	}
+}
+
+// Update the unbonding frequency of juno to 5 to align with the 28 day unbonding period
+func ModifyJunoUnbondingFrequency(ctx sdk.Context, k stakeibckeeper.Keeper) error {
+	ctx.Logger().Info("Updating juno unbonding frequency")
+
+	junoChainId := "juno-1"
+	unbondingFrequency := uint64(5)
+
+	junoHostZone, found := k.GetHostZone(ctx, junoChainId)
+	if !found {
+		return stakeibctypes.ErrHostZoneNotFound
+	}
+	junoHostZone.UnbondingFrequency = unbondingFrequency
+	k.SetHostZone(ctx, junoHostZone)
+
+	return nil
+}
+
+// Incentive diversification (Prop #153) - Send 3M STRD from Incentive Program to Stride Foundation
```

Comment:
```suggestion
// Incentive diversification (Prop #153) - Send 3M STRD from Incentive Program to Stride Foundation
// so that it can be swapped for USDC and deployed as incentives 
```

Reply:
Done